recOrder/cli/parsing.py

```diff
@@ -1,54 +1,75 @@
+
+from recOrder.cli.option_eat_all import OptionEatAll
 
 
-def _validate_fov_path(
+def _validate_and_process_paths(
     ctx: click.Context, opt: click.Option, value: str
 ) -> None:
-    dataset = open_ome_zarr(value)
-    if isinstance(dataset, Plate):
-        raise ValueError(
-            "Please supply a single position instead of an HCS plate. Likely fix: replace 'input.zarr' with 'input.zarr/0/0/0'"
-        )
-    return value
+    # Sort and validate the input paths
+    input_paths = [Path(path) for path in natsorted(value)]
+    for path in input_paths:
+        with open_ome_zarr(path, mode="r") as dataset:
+            if isinstance(dataset, Plate):
+                raise ValueError(
+                    "Please supply a single position instead of an HCS plate. Likely fix: replace 'input.zarr' with 'input.zarr/0/0/0'"
+                )
+    return input_paths
 
 
-def input_data_path_argument() -> Callable:
+def input_position_dirpaths() -> Callable:
     def decorator(f: Callable) -> Callable:
-        return click.argument(
-            "input-data-path",
-            type=click.Path(exists=True),
-            callback=_validate_fov_path,
-            nargs=1,
+        return click.option(
+            "--input-position-dirpaths",
+            "-i",
+            cls=OptionEatAll,
+            type=tuple,
+            callback=_validate_and_process_paths,
+            help="Paths to input positions",
         )(f)
 
     return decorator
 
 
-def config_path_option() -> Callable:
+def config_filepath() -> Callable:
     def decorator(f: Callable) -> Callable:
         return click.option(
-            "--config-path", "-c", required=True, help="Path to config.yml"
+            "--config-filepath",
+            "-c",
+            required=True,
+            type=click.Path(exists=True, file_okay=True, dir_okay=False),
+            help="Path to YAML configuration file",
         )(f)
 
     return decorator
 
 
-def output_dataset_option(default) -> Callable:
-    click_options = [
-        click.option(
-            "--output-path",
-            "-o",
-            default=default,
-            help="Path to output.zarr",
-        )
-    ]
-    # good place to add chunking, overwrite flag, etc
+def transfer_function_dirpath() -> Callable:
+    def decorator(f: Callable) -> Callable:
+        return click.option(
+            "--transfer-function-dirpath",
+            "-t",
+            required=True,
+            type=click.Path(exists=False, file_okay=False, dir_okay=True),
+            help="Path to transfer function .zarr",
+        )(f)
+
+    return decorator
+
 
+def output_dirpath() -> Callable:
     def decorator(f: Callable) -> Callable:
-        for opt in click_options:
-            f = opt(f)
-        return f
+        return click.option(
+            "--output-dirpath",
+            "-o",
+            required=True,
+            type=click.Path(exists=False, file_okay=False, dir_okay=True),
```

Comment:
Will `file_okay` and `dir_okay` do anything when `exists=False`?

Reply:
Good call, but I still think this is still useful for "append mode" where you set the output to an existing zarr store with the intent of adding to the reconstruction to that zarr store. 

We haven't implemented this yet, but I think it will be a valuable use case. Particularly since we have one config per reconstruction and we often want to do multiple reconstructions from a single zarr into a single zarr.

Reply:
When that feature is implemented we will have to edit this line again anyway to remove the `exists` check.

Reply:
Agreed! Thanks for your patience. Removed